### PR TITLE
Home frontend tab navigation

### PR DESCRIPTION
**Pull request by ghost**
### Summary
News/Profile tab navigation added to HomeActivity

### Card
https://trello.com/c/HXzxVcpM/15-u4-home-page-tabs-de-navegaci%C3%B3n

### Screen Recordings
homeTabNavigation

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/example/model/User.java
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/model/User.java]: Here is missing nullability

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt]: We can use a companion object for this

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt]: Replace {} with an =, so you can write it in a single line

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeNewsFragment.kt
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeNewsFragment.kt]: Same as above

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeProfileFragment.kt
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeProfileFragment.kt]: Same as above

File: app/src/main/res/layout/fragment_home.xml
Comment (lmiotti) [inline app/src/main/res/layout/fragment_home.xml:22]: Move this line above src

File: app/src/main/res/layout/fragment_home_news.xml
Comment (lmiotti) [inline app/src/main/res/layout/fragment_home_news.xml:5]: Solve this harcoded testin color

File: app/src/main/res/layout/fragment_home_profile.xml
Comment (lmiotti) [inline app/src/main/res/layout/fragment_home_profile.xml:5]: Same as above

File: app/src/main/res/layout/fragment_home.xml
Comment (lmiotti) [inline app/src/main/res/layout/fragment_home.xml:50]: Be careful when you use wrap_content on a height attribute in viewPager. We can leave it now as it is. Maybe we have to change it in following PR

File: app/src/main/res/layout/fragment_home.xml
Comment (lmiotti) [inline app/src/main/res/layout/fragment_home.xml]: Be careful with attributes order. Layout dimens modifications first and then text/src/etc attributes

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt]: In Kotlin we are not going to use 'm' as a prefix

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt:17]: Add an empty line to separate injections with local vars

**Review by lmiotti — CHANGES_REQUESTED** [review]
It's quite good. Correct the things I've marked and we're done!

File: app/src/main/java/ar/com/wolox/android/example/model/User.java
Comment (ghost) [reply app/src/main/java/ar/com/wolox/android/example/model/User.java]: Yes, I don't put nullability here because is a built-in data type **int**, I thought that only have sense to put nullability for objects, for example if it was an **Integer**. Tell me what you think!

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt
Comment (ghost) [reply app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt]: ✔ **Done**

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeNewsFragment.kt
Comment (ghost) [reply app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeNewsFragment.kt]: ✔ **Done**

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeProfileFragment.kt
Comment (ghost) [reply app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeProfileFragment.kt]: ✔ **Done**

File: app/src/main/res/layout/fragment_home.xml
Comment (ghost) [reply app/src/main/res/layout/fragment_home.xml:22]: ✔ **Done**

File: app/src/main/res/layout/fragment_home_news.xml
Comment (ghost) [reply app/src/main/res/layout/fragment_home_news.xml:5]: This is hardcoded because is the color of the background used for testing that news and profile pages are changing from one to another, will be removed on the next PR where I will create the layout for each pages. What do you think?

File: app/src/main/res/layout/fragment_home_profile.xml
Comment (ghost) [reply app/src/main/res/layout/fragment_home_profile.xml:5]: Same answer as above

File: app/src/main/res/layout/fragment_home.xml
Comment (ghost) [reply app/src/main/res/layout/fragment_home.xml:50]: Ok, I will took in account!

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt
Comment (ghost) [reply app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt]: ✔ **Changed**

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt
Comment (ghost) [reply app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt:17]: ✔ **Done**

File: app/src/main/res/layout/fragment_home.xml
Comment (ghost) [reply app/src/main/res/layout/fragment_home.xml]: ✔ **Done**

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt
Comment (ghost) [reply app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt]: ✔ **Done:** companion wrapping an enum class

**Comment by ghost** [comment]
Changes done except for the ones I commented above. Waiting for review!

File: app/src/main/java/ar/com/wolox/android/example/model/User.java
Comment (lmiotti) [reply app/src/main/java/ar/com/wolox/android/example/model/User.java]: Yes, you're absolutely right. If it's a built-in type you don't need a nullability. My bad

File: app/src/main/res/layout/fragment_home_news.xml
Comment (lmiotti) [reply app/src/main/res/layout/fragment_home_news.xml:5]: Ok, I understand. We are going to leave it and change it in future

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt]: Move companion object to the end of the file

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt:28]: It's necessary to use index? If using TabIndex.NEWS it will give a 0 due to the fact that you created NEWS(0)

**Review by lmiotti — CHANGES_REQUESTED** [review]
It's almost done. Tell me what you think!

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt
Comment (ghost) [reply app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt:28]: Yes, I thought the same, but apparently is necessary.

image

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt
Comment (ghost) [reply app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeFragment.kt]: ✔ **Done**

**Comment by ghost** [comment]
Perfect. Waiting for approval! :smile:

**Review by lmiotti — APPROVED** [review]
james_corden

**Pull request merged**
